synapse/server_notices/resource_limits_server_notices.py

```diff
@@ -135,7 +135,7 @@ async def _remove_limit_block_notification(
         )
 
     async def _apply_limit_block_notification(
-        self, user_id: str, event_body: str, event_limit_type: str
+        self, user_id: str, event_body: str, event_limit_type: Optional[str]
```

Comment:
I'm not sure if I see much evidence of `event_limit_type` ever being set to `None` by the calling function?

Reply:
https://github.com/matrix-org/synapse/blob/4a3f15deaca4259915af4413dfb2acf91fdc9a98/synapse/server_notices/resource_limits_server_notices.py#L88-L97

`limit_type` only gets set on error.

Reply:
(I should also note that mypy found this.)

Reply:
It looks like in the linked code block that `_apply_limit_block_notification` will only be called if `limit_msg` is not `None`, and as `limit_msg` and `limit_type` are both set together, `_apply_limit_block_notification` should only be called when `limit_type` is not `None`?

I suppose that may just be circumstantial of this one calling function though, and `_apply_limit_block_notification` can handle `limit_type=None` just fine, and thus that's what the signature should be. I found it a bit difficult to verify that last bit though...

Reply:
Ah, I see what you're saying. Let me see if I can convince mypy here...

Reply:
Yeah so mypy doesn't seem to recognize that these are both set or neither is set. I switched it around a bit in 052a474a54d2797e23ffe758ae6d547bf2ae5d68, but ended up ignoring that function call. I believe the correct signature is `str`, not `Optional[str]`, so I'd rather the signature be correct.

Let me know if you agree!